core/src/main/java/org/elasticsearch/percolator/PercolateDocumentParser.java

```diff
@@ -78,8 +75,8 @@ public ParsedDocument parse(PercolateShardRequest request, PercolateContext cont
         // Some queries (function_score query when for decay functions) rely on a SearchContext being set:
         // We switch types because this context needs to be in the context of the percolate queries in the shard and
         // not the in memory percolate doc
-        String[] previousTypes = context.types();
-        context.types(new String[]{PercolatorService.TYPE_NAME});
+        String[] previousTypes = context.getQueryShardContext().getTypes();
```

Comment:
Maybe not for this PR since the method signature hasn't changed but I don't think we need to pass in the QueryShardContext in the method parameters since it's in the PercolateContext anyway?